### DISCUSSION TITLE
Is it ready for Sequoia

### DISCUSSION THREAD

**Discussion by imkebe** [rxhanson/RectanglePro-Community#629] [General]
Sequoia is getting a functionality like Rectangle, however it is still basic one. Will the Rectangle Pro conflicting the Seqioua built-in feature?

**Comment by rxhanson** [comment]
If you have the drag to screen edge snapping enabled in Rectangle Pro, then that will conflict with the built-in functionality in Sequoia. Rectangle Pro will prompt on the first run if you want to disable either one. The Mac settings are in System Settings > Desktop & Dock > Windows.

**Comment by seahindeniz** [comment]
@rxhanson is it possible to get that prompt again somehow? I chose the option wrong, and now it's a torture 😞

**Comment by rxhanson** [comment]
Which option are you looking to disable? 

macOS:
![Untitled](https://github.com/user-attachments/assets/2c939705-2953-419a-b849-79a7279f0c95)

Rectangle Pro:
![image](https://github.com/user-attachments/assets/ef8418e2-0208-4d7c-8c50-a941c305deec)

**Comment by JoePearlAdobe** [comment]
@rxhanson I have been super frustrated lately with my window layout.  I'm on Sequoia 15.3 and I have Rectangle Pro 3.0.41 (182).  I've turned off the MacOS tiling functionality as your screen shot shows.  I'm getting closer to what I want but not quite there.  I imagine I'm missing something in the Rectangle Pro settings.

I have added 3 different layouts for the way I like my windows of my applications to be arranged.  I have an MBP 16" November 2023 with the M3 Max processor.  I have an external monitor connected through a USB-C port.  

If I restart my Mac and do not re-open applications, I'm getting the best results.  The two layouts on the external monitor seem to be working most of the time, apparently only when I choose a desktop other than 1 (my large monitor is my primary).  My browser, which I want to start in a different desktop and on the MacBook, won't start.  I use Arc.  I've tried starting using the shortcut I created as well as choosing it from the Rectangle Pro menu.

I do not get any of the layouts to restore with "Trigger on display connected" 

Any thoughts?

![image](https://github.com/user-attachments/assets/137ae77b-4a11-4ff2-ab8f-3366aae4f547)

**Reply by JoePearlAdobe** [reply]
Update - I forgot to check the "launch application" for starting Arc.  

Are the layouts tied directly to the specific desktop?

**Reply by rxhanson** [reply]
Triggering on display connected is a surprisingly difficult one to get right, and it's because timing on some displays can be unpredictable and really long. Because of this, Rectangle Pro will attempt to position windows upon display connection after a short delay and if that fails it doesn't keep trying. When I've tested retry mechanisms to work around this, the end result can be a worse user experience with windows being positioned at a delay. I don't mind giving this another try and seeing if I can figure out a better solution. 

> Are the layouts tied directly to the specific desktop?

I think you're referring to spaces? Layouts are not tied to Spaces, as there is no public API for Spaces.

**Reply by JoePearlAdobe** [reply]
Thanks for the response Ryan.  IMHO, the trigger on display connecting isn't a major problem.  Yes, it was Spaces that i was referencing.  I figured out that if I boot the Mac without restoring applications, then choose a space, open my saved Rectangle Pro layout, I can get what I need.  My primary pain was EVERY window/application starting on Space 1, Display 1 on a reboot.  Using Rectangle Pro and the layouts I've created, I have saved a lot of time and frustration.
